rust/pattern/constraint/type_/owns.rs

```diff
@@ -89,113 +79,314 @@ impl Validatable for OwnsConstraint {
 
 impl From<&str> for OwnsConstraint {
     fn from(attribute_type: &str) -> Self {
-        OwnsConstraint::from((attribute_type, IsKeyAttribute::No))
+        OwnsConstraint::from(Label::from(attribute_type))
     }
 }
 
-impl From<(&str, IsKeyAttribute)> for OwnsConstraint {
-    fn from((attribute_type, is_key): (&str, IsKeyAttribute)) -> Self {
-        OwnsConstraint::from((Label::from(attribute_type), is_key))
+impl From<String> for OwnsConstraint {
+    fn from(attribute_type: String) -> Self {
+        OwnsConstraint::from(Label::from(attribute_type))
     }
 }
 
-impl From<(String, IsKeyAttribute)> for OwnsConstraint {
-    fn from((attribute_type, is_key): (String, IsKeyAttribute)) -> Self {
-        OwnsConstraint::from((Label::from(attribute_type), is_key))
+impl From<Label> for OwnsConstraint {
+    fn from(attribute_type: Label) -> Self {
+        OwnsConstraint::from(UnboundVariable::hidden().type_(attribute_type))
     }
 }
 
-impl From<(Label, IsKeyAttribute)> for OwnsConstraint {
-    fn from((attribute_type, is_key): (Label, IsKeyAttribute)) -> Self {
-        OwnsConstraint::from((UnboundVariable::hidden().type_(attribute_type), is_key))
+impl From<Type> for OwnsConstraint {
```

Comment:
After investigation we found that `Label.Type` shouldn't exist outside of the Parser implementation, which will also require that the constructors here for `Type` will disappear.